### PR TITLE
Add PagerdutyURL to AlertmanagerGlobalConfig

### PR DESCRIPTION
**Pull request by mcbenjemaa**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

Add PagerdutyURL to AlertmanagerGlobalConfig

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [X] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Adds  `spec.alertmanagerConfiguration.global.PagerdutyURL`  to the `Alertmanager` CRD
```

### CODE REVIEW

**Review by JoaoBraveCoding — APPROVED** [review]
/lgtm

File: pkg/apis/monitoring/v1/alertmanager_types.go [hunk 1]
```diff
@@ -282,6 +282,9 @@ type AlertmanagerGlobalConfig struct {
 
 	// The default OpsGenie API Key.
 	OpsGenieAPIKey *v1.SecretKeySelector `json:"opsGenieApiKey,omitempty"`
+
+	// The default Pagerduty URL.
+	PagerdutyURL *v1.SecretKeySelector `json:"pagerdutyUrl,omitempty"`
```
[the remark this answers is not in the document]
Comment (JoaoBraveCoding) [reply pkg/apis/monitoring/v1/alertmanager_types.go:287]: Yeah, I followed that same logic when I reviewed it 🤔

[the remark this answers is not in the document]
Comment (simonpasquier) [reply pkg/apis/monitoring/v1/alertmanager_types.go:287]: AFAICT there's nothing sensitive here (at least it's marked as such in the Alertmanager configuration struct).
I'm not even sure why this is configurable as the URL should be `https://events.pagerduty.com/v2/enqueue` .

[the remark this answers is not in the document]
Comment (simonpasquier) [reply pkg/apis/monitoring/v1/alertmanager_types.go:287]: anyway it's a plain string in the Alertmanager config so we need a string.

[the remark this answers is not in the document]
Comment (JoaoBraveCoding) [reply pkg/apis/monitoring/v1/alertmanager_types.go:287]: > at least it's marked as such in the Alertmanager configuration struct

I see I should have checked, that! Indeed, looks like a string! I'll be more careful next time!

**Comment by mcbenjemaa** [comment]
Closing this in favor: https://github.com/prometheus-operator/prometheus-operator/pull/5469

**Pull request closed without merging**